### PR TITLE
po: sync translations from launchpad

### PR DESCRIPTION
**Pull request by ZeyadYasser**
- Sync translations from Launchpad
- Add `po/README.md` as a small guide for future updates

### CODE REVIEW

File: po/arn.po
Comment (MiguelPires) [inline po/arn.po:1]: Some of these files are quite large but seem to contain very few translations. Scrolling around at random, I could only find 2 translations in this one, for instance. I wonder if there's a way to export just non-empty translations? I imagine that would make this PR a lot smaller.

This says something about merging partial exports https://help.launchpad.net/Translations/YourProject/PartialPOExport but it seems to be about just exporting the delta between files. Not sure if there's a way to skip all the blocks with empty translations. Maybe @mvo5 knows more

**Review by MiguelPires — COMMENTED** [review]
Thanks for this. These files seem to have mostly empty translations, left a comment about that (not sure if it's something we can easily improve on)

File: po/arn.po
Comment (ZeyadYasser) [reply po/arn.po:1]: Thank you for pointing this out. I think we could use [msgattrib](https://linux.die.net/man/1/msgattrib) cmd (i.e. `--translated` and `--no-obsolete`) to do some preprocessing before pushing to get rid of unnecessary sections. I will look into it and update the PR.

Comment (ZeyadYasser) [reply po/arn.po:1]: Removed obsolete comments and untranslated sections, this made the PR much smaller. Thank you!

File: po/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- Run `update-pot` to generate latest translation template (`po/snappy.pot`).
+- Import generated translation template to snappy entry [here](https://translations.launchpad.net/snapd/trunk/+templates).
+- Wait until import is approved. You can monitor status in the import queue [here](https://translations.launchpad.net/snapd/trunk/+imports).
+- Export latest translations [here](https://translations.launchpad.net/snapd/trunk/+export) (download link will be sent to your email).
+- Extract archive: `tar -xvf launchpad-export.tar.gz`.
+- Create new directory to store processed translations `mkdir processed`.
+- `for file in snappy-*; do msgattrib --no-obsolete --translated --output-file="processed/${file#snappy-}" "$file"; done;`.
+    - Remove `snappy-` prefix.
+    - Remove obsolete #~ messages  `--no-obsolete`.
+    - Keep translated, remove untranslated messages `--translated`.
```
Comment (MiguelPires) [inline po/README.md:13]: Minor nitpick, reads slightly better like this to me:

```suggestion
    - Removes the `snappy-` prefix
    - `--no-obsolete` removes obsolete #~ messages
    - `--translated` excludes untranslated messages
```

**Review by MiguelPires — APPROVED** [review]
Nice, thanks for this

File: po/README.md (hunk 1 above)
Comment (ZeyadYasser) [reply po/README.md]: Agreed, updated

**Comment by MiguelPires** [comment]
You might've seen it in the [CONTRIBUTING.md](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md) document but we try not to force push to PRs once they started getting reviews. It's not a big deal in this case but usually it makes it difficult for reviewers to see what changed between pushes because the later push re-wrote history

**Comment by mvo5** [comment]
Thanks for this! I think there is also a way for launchpad to export translations as git repositories (or maybe just via bzr?), have you looked into this? That might make merging simpler?

**Comment by ZeyadYasser** [comment]
@mvo5 Yes, launchpad only supports automatic exports to bazaar branches according to [this guide](https://help.launchpad.net/Translations/YourProject/Exports#Committing_translations_to_a_Bazaar_branch). For example, automatic exports is already setup in [snappy project](https://translations.launchpad.net/snappy/trunk/+translations-settings) on the [snappy-moved-to-github](https://bazaar.launchpad.net/~snappy-dev/snappy/snappy-moved-to-github/files/head:/po?sort=date) branch. But it seems to be limited to bazaar :(

**Comment by ZeyadYasser** [comment]
Updated commit author and message to match Launchpad automated commits.

**Review by olivercalder — APPROVED** [review]
LGTM, though the CLA check is failing. Does "Launchpad Translations on behalf of snappy-dev" need to sign the CLA? :)

**Comment by ZeyadYasser** [comment]
> LGTM, though the CLA check is failing. Does "Launchpad Translations on behalf of snappy-dev" need to sign the CLA? :)

@olivercalder I am not sure if it even makes to associate an email with "Launchpad Translations" as the author.

**Comment by MiguelPires** [comment]
> LGTM, though the CLA check is failing. Does "Launchpad Translations on behalf of snappy-dev" need to sign the CLA? :)

This needs some config changes so the CLA check passes. I'll fix that and then merge this

**Comment by ZeyadYasser** [comment]
According the [CLA check action](https://github.com/canonical/has-signed-canonical-cla/blob/9a7e0da38a13dbc25b14c389851bcf1624f4784d/index.js#L63) source code, I removed the extra space from the author name.
The check is still failing because the CLA action checks the [github username](https://github.com/canonical/has-signed-canonical-cla/blob/9a7e0da38a13dbc25b14c389851bcf1624f4784d/index.js#L44C33-L44C52) (i.e. `login` field of the API response) of the committer (which only exists for real github bots), but doesn't in our case since this is not a registered bot.
The bot exemption is bypassed due to [empty github username](https://github.com/canonical/has-signed-canonical-cla/blob/9a7e0da38a13dbc25b14c389851bcf1624f4784d/index.js#L60).
I am guessing that the CLA check action needs to be updated to account for such case so it also checks for the commit author name as well when testing for bot exemptions and not only the login username.

**Comment by ernestl** [comment]
Hi @ZeyadYasser, this seems to have stalled, what is the next step?

**Comment by ernestl** [comment]
Discussed with Zeyad, this is now outdated. Intend to commit the useful information in the readme in seperate PR (without CLA issues). Closing.

**Pull request closed without merging**